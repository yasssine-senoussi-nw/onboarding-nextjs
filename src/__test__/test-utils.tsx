import React from "react";

import { TestTranslationProvider } from "~i18n";

// eslint-disable-next-line no-restricted-imports
import type {
  Queries,
  queries,
  RenderHookOptions,
  RenderHookResult,
  RenderOptions,
  RenderResult,
} from "@testing-library/react";
// eslint-disable-next-line no-restricted-imports
import { render, renderHook } from "@testing-library/react";
import type { ReactNode } from "react";

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return <TestTranslationProvider>{children}</TestTranslationProvider>;
};

const customRender = <
  Q extends Queries = typeof queries,
  Container extends DocumentFragment | Element = HTMLElement,
  BaseElement extends DocumentFragment | Element = Container,
>(
  ui: ReactNode,
  options?: RenderOptions<Q, Container, BaseElement>,
): RenderResult<Q, Container, BaseElement> => render(ui, { wrapper: AllTheProviders, ...options });

const customRenderHook = <
  Result,
  Props,
  Q extends Queries = typeof queries,
  Container extends DocumentFragment | Element = HTMLElement,
  BaseElement extends DocumentFragment | Element = Container,
>(
  hook: (initialProps: Props) => Result,
  options?: RenderHookOptions<Props, Q, Container, BaseElement>,
): RenderHookResult<Result, Props> => {
  return renderHook<Result, Props, Q, Container, BaseElement>(hook, { wrapper: AllTheProviders, ...options });
};

// eslint-disable-next-line no-restricted-imports
export * from "@testing-library/react";
// eslint-disable-next-line no-restricted-imports
export * from "@testing-library/user-event";
export { customRender as render, customRenderHook as renderHook };
