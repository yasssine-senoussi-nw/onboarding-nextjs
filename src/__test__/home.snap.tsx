import HomePage from "~app/page";

import { render } from "./test-utils";
jest.mock("next/navigation");

describe("Testing page example", () => {
  it("should render home", () => {
    const { container } = render(<HomePage />);
    expect(container).toMatchSnapshot();
  });
});
