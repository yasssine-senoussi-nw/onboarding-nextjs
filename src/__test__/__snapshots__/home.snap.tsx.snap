// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing page example should render home 1`] = `
.emotion-1 {
  background-color: black;
  background-image: url(/assets/svg/background.svg);
  -webkit-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
  height: 100vh;
  width: 100wh;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  min-width: 659px;
  max-width: 750px;
}

.emotion-3 {
  margin-bottom: 32px;
  width: 295px;
  height: 63px;
}

.emotion-4 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: white;
  text-align: center;
  font-size: 95px;
  line-height: 100px;
  font-weight: 600;
  margin-bottom: 35px;
}

.emotion-5 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.5;
  letter-spacing: 0.00938em;
  color: white;
  text-align: center;
  font-weight: 500;
  line-height: 20px;
  font-size: 18px;
  margin-bottom: 33px;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-6>:not(style)+:not(style) {
  margin: 0;
  margin-left: 18px;
}

.emotion-7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  width: 8.625rem;
  height: 3.5rem;
  padding: 0.625rem 0.813rem;
  border-radius: 3.75rem;
  font-size: 1.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.625rem;
  border-width: 0.125rem;
  border-style: solid;
  border-color: #fff;
  background-color: transparent;
  font-weight: 700;
  color: #fff;
}

.emotion-7::-moz-focus-inner {
  border-style: none;
}

.emotion-7.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-7 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-7:hover {
  background-color: rgba(255, 255, 255, .12);
}

.emotion-8 {
  overflow: hidden;
  pointer-events: none;
  position: absolute;
  z-index: 0;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  border-radius: inherit;
}

.emotion-9 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  position: relative;
  box-sizing: border-box;
  -webkit-tap-highlight-color: transparent;
  background-color: transparent;
  outline: 0;
  border: 0;
  margin: 0;
  border-radius: 0;
  padding: 0;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  vertical-align: middle;
  -moz-appearance: none;
  -webkit-appearance: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  width: 8.625rem;
  height: 3.5rem;
  padding: 0.625rem 0.813rem;
  border-radius: 3.75rem;
  font-size: 1.125rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 0.625rem;
  background-color: white;
  color: #9c27b0;
  font-weight: 700;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
}

.emotion-9::-moz-focus-inner {
  border-style: none;
}

.emotion-9.Mui-disabled {
  pointer-events: none;
  cursor: default;
}

@media print {
  .emotion-9 {
    -webkit-print-color-adjust: exact;
    color-adjust: exact;
  }
}

.emotion-9:hover {
  background-color: #42a5f5;
  color: #fff;
}

<div>
  <div
    class="MuiBox-root emotion-0"
  >
    <div
      class="landing emotion-1"
    >
      <div
        class="MuiBox-root emotion-2"
      >
        <img
          alt="Theodo"
          class="emotion-3"
          data-nimg="1"
          decoding="async"
          height="40"
          loading="lazy"
          src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
          srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
          style="color: transparent;"
          width="40"
        />
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-4"
        >
          Outils triés pour vous équiper !
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 emotion-5"
        >
          Rejoignez l’initiative Worktools de Nimbleway pour être bien préparé à vos tâches quotidiennes.
        </p>
        <div
          class="emotion-6"
        >
          <button
            class="MuiButtonBase-root emotion-7"
            tabindex="0"
            type="button"
          >
            Rejoindre
            <span
              class="MuiTouchRipple-root emotion-8"
            />
          </button>
          <button
            class="MuiButtonBase-root emotion-9"
            tabindex="0"
            type="button"
          >
            S'inscrire
            <span
              class="MuiTouchRipple-root emotion-8"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
